.github/workflows/test_cpu.yml

```diff
@@ -0,0 +1,79 @@
+name: Run tests on CPU
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+    types: [open, reopened, synchronize, review_requested, ready_for_review]
+  schedule:
+    - cron: "0 4 * * *"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  tests-cpu:
+    name: ${{ matrix.os }} - python-${{ matrix.python-version }}, torch-${{ matrix.pytorch-version }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: ['3.7', '3.8']
+        pytorch-version: ['lts', '1.10.2', '1.12.1', 'nightly']
```

Comment:
Based on slack discussion should we still target lts?

Reply:
I will remove it, do you think we should go further and leave only the last two minor versions and nightly?

Reply:
sure, for the experience `nightly` is unstable all the time so i'd suggest to schedule that build instead of requiring in each PR? also we could include 1.9.1 in a scheduled matrix

Reply:
Makes sense to me, I will split the workflows, build one just for PR's, and another for the main

Reply:
perfect